packages/next/cli/next-dev.ts

```diff
@@ -106,15 +106,12 @@ const nextDev: cliCommand = (argv) => {
   }
 
   const port = args['--port'] || 3000
-  const appUrl = `http://${args['--hostname'] || 'localhost'}:${port}`
+  const host = args['--hostname'] || '0.0.0.0'
+  const appUrl = `http://${host === '0.0.0.0' ? '127.0.0.1' : host}:${port}`
```

Comment:
To mirror our current output:
```suggestion
  const appUrl = `http://${host === '0.0.0.0' ? 'localhost' : host}:${port}`
```

Otherwise people might think we started listening on another host and start creating bug reports for it.

Reply:
Done!